system/Debug/Toolbar.php

```diff
@@ -374,7 +374,7 @@ public function prepare(?RequestInterface $request = null, ?ResponseInterface $r
             helper('filesystem');
 
             // Updated to time() so we can get history
-            $time = time();
+            $time = number_format(microtime(true), 4, '.', '');
```

Comment:
Do you really need the 4 decimals?

Reply:
It is important to note that microtime(TRUE) does NOT always return a float, with the same number of decimals (php.ini can also affect microtime output).  
If it happens to land on an exact second, it returns an integer instead.

Here is some simple code to illustrate this:
```php
while(true) {
	usleep(100);
	$mt = microtime(TRUE);
	if(!str_contains($mt, ".")) {   /// str_contains only for php > 8.0
		echo $mt;
		break;
	}
}
```


When you run this at a command line, you'll see output like this (and you'll know EXACTLY when I ran this):

1643869315.6995 
1643869315.6996 
1643869315.6998 
1643869315          <-- 
1643869315.7001 
1643869315.7003 
1643869315.7005

Why does this matter?  Because if you want to format microtime into a very friendly but exact date/time, this popular code will fail on exact seconds (but works great otherwise):

```php
   $raw_time = DateTime::createFromFormat('U.u', microtime(TRUE)); // This line will not create an object when microtime returns an integer because it depends on the number being a float, including a decimal point!
   $today = $raw_time->format('Y-m-d H:i:s.u'); // This will die when format is attempted on a non-object
```

The fix is to format the microtime output to force it to always be a float:
```php
   $utime = sprintf('%.4f', microtime(TRUE)); /// or use number_format
   $raw_time = DateTime::createFromFormat('U.u', $utime);
   $today = $raw_time->format('Y-m-d H:i:s.u');
```

Reply:
I think the easiest solution here is `$time = microtime(true) * 10000;`.
This should give us the microtime as an integer value represnting microseconds in any of the cases you mentioned. It will also avoid the BC breaks in the method signatures.

Reply:
On second thought this might not work as it exceeds the `int` limit.